src/mesh/meshblock.cpp

```diff
@@ -179,25 +179,25 @@ void MeshBlock::Initialize(int igid, int ilid, LogicalLocation iloc,
   // potentially used in the load balancing calculation, but not all
   // variables that we may want to communicate are necessarily relevant
   // to the cost per meshblock.
-  const auto vars = real_container
-                        ->GetVariablesByFlag({Metadata::Independent, Metadata::FillGhost,
-                                              Metadata::ForceRemeshComm},
-                                             false)
-                        .vars();
-  for (int n = 0; n < vars.size(); ++n) {
-    RegisterMeshBlockData(vars[n]);
+  using FS_t = Metadata::FlagCollection;
+  // union of Independent and FillGhost
+  FS_t flags({Metadata::Independent, Metadata::FillGhost}, true);
+  // Toss in RemeshComm for this one
+  const auto vars =
+      real_container->GetVariablesByFlag(flags + FS_t({Metadata::ForceRemeshComm}, true))
+          .vars();
+  for (const auto &v : vars) {
+    RegisterMeshBlockData(v);
   }
 
+  // No RemeshComm
   if (pm->multilevel) {
-    const auto refine_vars =
-        real_container
-            ->GetVariablesByFlag({Metadata::Independent, Metadata::FillGhost}, false)
-            .vars();
+    const auto refine_vars = real_container->GetVariablesByFlag(flags).vars();
     pmr = std::make_unique<MeshRefinement>(shared_from_this(), pin);
     // This is very redundant, I think, but necessary for now
-    for (int n = 0; n < refine_vars.size(); n++) {
+    for (const auto &v : vars) {
```

Comment:
Why is this now going over `vars` and not `refine_vars`?

Reply:
This is a bug. Not caught because I don't think we have any tests in `parthenon` where `ForceRemeshComm` is present. Fixed now.